File: server/src/models/Teacher.js
const mongoose = require('mongoose');

const teacherSchema = new mongoose.Schema({
  name: { type: String, required: true },
  title: { type: String, required: true },
  location: { type: String, required: true },
  rating: { type: Number, default: 0 },
  about: { type: String },
  isOnline: { type: Boolean, default: false },
  studentsCount: { type: Number, default: 0 },
  lessonsCount: { type: Number, default: 0 },
  hoursCount: { type: Number, default: 0 },
  status: { type: String, default: 'pending' },
  tags: [String],
  sessionTopics: [String],
  timeSlots: [{
    start: String,
    end: String
  }],
});

module.exports = mongoose.model('Teacher', teacherSchema);

File: client/lib/models/teacher.dart
class Teacher {
  final String id;
  final String name;
  final String email;
  final String status;

  Teacher({
    required this.id,
    required this.name,
    required this.email,
    required this.status,
  });

  factory Teacher.fromJson(Map<String, dynamic> json) {
    return Teacher(
      id: json['_id'] ?? json['id'],
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      status: json['status'] ?? 'pending',
    );
  }
}
